abci/state/deposit/deposit.go

```diff
@@ -0,0 +1,83 @@
+package deposit
+
+import (
+	"github.com/likecoin/likechain/abci/account"
+	"github.com/likecoin/likechain/abci/context"
+	"github.com/likecoin/likechain/abci/response"
+	"github.com/likecoin/likechain/abci/types"
+)
+
+// GetDepositApproverInfo checks if a LikeChain ID is one of the DepositApprovers
+func GetDepositApproverInfo(state context.IImmutableState, id *types.LikeChainID) *Approver {
+	depositApprovers := GetDepositApprovers(state)
+	for _, approver := range depositApprovers {
+		if id.Equals(approver.ID) {
+			return &approver
+		}
+	}
+	return nil
+}
+
+// CheckDepositProposal checks if a proposer could be proposed in the current context
+func CheckDepositProposal(state context.IImmutableState, proposal Proposal, proposer *types.LikeChainID) response.R {
+	blockNumber := proposal.BlockNumber
+	if GetDepositExecution(state, blockNumber) != nil {
+		return response.DepositAlreadyExecuted
+	}
+	if GetDepositApproverInfo(state, proposer) == nil {
+		return response.DepositNotApprover
+	}
+	if GetDepositApproval(state, proposer, blockNumber) != nil {
+		return response.DepositDoubleApproval
+	}
+	return response.Success
+}
+
+// CreateDepositProposal creates a deposit proposal and related entries in the context, returns the weight of the proposal
+func CreateDepositProposal(state context.IMutableState, txHash []byte, proposal Proposal, proposer *types.LikeChainID) uint64 {
```

Comment:
Should use input hash instead of txHash as key of proposal, otherwise the same proposal could be proposed multiple times, blocking each others